Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -356,6 +356,13 @@ void RenderBox::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle
         }
     }
 
+#if ENABLE(DARK_MODE_CSS)
+    if (layer() && oldStyle && oldStyle->colorScheme() != newStyle.colorScheme()) {
+        if (auto* scrollableArea = layer()->scrollableArea())
+            scrollableArea->invalidateScrollbars();
```

Comment:
I tried also adding `scrollableArea->invalidateScrollCornerRect(scrollableArea->scrollCornerRect());` here but that didn't fix the scroll corner not updating.

Reply:
May be worth looking into if `LocalFrameView::paintScrollCorner` gets called, and checking the value of `useDarkAppearanceForScrollbars()`.

Reply:
I've found the fix for non-viewport scroll corners (when scrollbars are rendered). I needed invalidateScrollCorner rather than calling invalidateScrollCornerRect